src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,38 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_PASSWORD_CHARACTERS = 6;
+    private static final int MIN_LOGIN_CHARACTERS = 6;
+
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        validation(user);
+        return storageDao.add(user);
+    }
+
+    private void validation(User user) {
```

Comment:
Method names should be verbs for better readability
```suggestion
    private void validate(User user) {
```